src/adapter/scriptSkipper.ts

```diff
@@ -59,26 +55,38 @@ export class ScriptSkipper {
   }
 
   private _updateBlackboxedUrls(url: string) {
-    if (this._isUrlSkippedMap.get(url) && this.blackboxSender) {
-      this._blackboxedUrls.push(url);
-      let blackboxPattern = '^' + this._blackboxedUrls.join('|') + '$';
+    if (this.blackboxSender) {
+      if (this._isUrlSkippedMap.get(url)) {
+        this._blackboxedUrls.push(url);
+      } else {
+        this._blackboxedUrls = this._blackboxedUrls.filter(blackBoxedUrl => blackBoxedUrl !== url);
+      }
+      let blackboxPattern = '^' + this._createRegexString(this._blackboxedUrls) + '$';
```

Comment:
I think you might need parens around the regex string, or the ^ and $ will just apply to the first and last patterns?